src/gate.ts

```diff
@@ -139,9 +142,19 @@ export class Gate {
                 "https://"+ this.t7.conf.net.peerbook)
         } else {
             editHost = document.getElementById("edit-host")
-            editHost.querySelector('[name="hostaddr"]').value = this.addr
-            editHost.querySelector('[name="hostname"]').value = this.name
-            editHost.querySelector('[name="username"]').value = this.username
+            if (editHost.classList.contains("hidden")) {
+                const e = editHost.querySelector(".terminal-container")
+                const t = openFormsTerminal(e)
+                const f = new Form([{ prompt: "Name", default: this.name, validator: Gate.validateHostName }, { prompt: "Hostname", default: this.addr }, { prompt: "Username", default: this.username }])
```

Comment:
This line is too long and too ugly. At the least, each field should get its own line and it's probably better to move it to the class level.